include/libopencm3/pac55xx/gpio.h

```diff
@@ -0,0 +1,334 @@
+ *  @date 1 Dec 2019
+ *
+ *  Definitions in this file come from the PAC55XX Family User Guide Rev 1.21
+ *  by Active-Semi dated August 26, 2019.
+ *
+ *  @defgroup PAC55xx_gpio_api GPIO API
+ *  @ingroup PAC55xx_gpio
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef INCLUDE_LIBOPENCM3_PAC55XX_GPIO_H_
+#define INCLUDE_LIBOPENCM3_PAC55XX_GPIO_H_
+
+#include <libopencm3/cm3/common.h>
+
+/* Note: CCS include file defines pinmux, drive strength, pull-up/pull-down, etc. */
+#include <libopencm3/pac55xx/ccs.h>
+#include <libopencm3/pac55xx/memorymap.h>
+
+/* GPIO port base addresses (for convenience) */
+/** @defgroup gpio_port_id GPIO Port IDs
+@ingroup PAC55xx_gpio
+@{*/
+#define GPIOA GPIOA_BASE
+#define GPIOB GPIOB_BASE
+#define GPIOC GPIOC_BASE
+#define GPIOD GPIOD_BASE
+#define GPIOE GPIOE_BASE
+#define GPIOF GPIOF_BASE
+#define GPIOG GPIOG_BASE
+/**@}*/
+
+/* GPIO number definitions (for convenience) */
+/** @defgroup gpio_pin_id GPIO Pin Identifiers
+@ingroup PAC55xx_gpio
+@{*/
+#define GPIO_MAX_PIN (7U)
+#define GPIO0 BIT0
+#define GPIO1 BIT1
+#define GPIO2 BIT2
+#define GPIO3 BIT3
+#define GPIO4 BIT4
+#define GPIO5 BIT5
+#define GPIO6 BIT6
+#define GPIO7 BIT7
+#define GPIO_ALL 0xff
+/**@}*/
+
+/* GPIO Mode Register Definitions */
+/** @defgroup gpio_mode_regs GPIO MODE register mapping.
+@ingroup PAC55xx_gpio
+@{*/
+/* Enum definitions for at least minimal type safety. */
+typedef enum {
+	GPIO_ANALOG_INPUT = 0,
+	GPIO_PUSH_PULL_OUTPUT = 1,
+	GPIO_OPEN_DRAIN_OUTPUT = 2,
+	GPIO_HIGH_IMPEDENCE_INPUT = 3,
+} gpio_mode_t;
+#define GPIO_MODER(base) MMIO32((base) + 0x00)
+#define GPIOA_MODER GPIO_MODER(GPIOA)
+#define GPIOB_MODER GPIO_MODER(GPIOB)
+#define GPIOC_MODER GPIO_MODER(GPIOC)
+#define GPIOD_MODER GPIO_MODER(GPIOD)
+#define GPIOE_MODER GPIO_MODER(GPIOE)
+#define GPIOF_MODER GPIO_MODER(GPIOF)
+#define GPIOG_MODER GPIO_MODER(GPIOG)
+#define GPIO_MODER_MASK 0x3
+#define GPIO_MODER_MASK_PIN(pin) (GPIO_MODER_MASK << ((pin)*2))
+#define GPIO_MODER_MODE(pin, mode) (((mode)&GPIO_MODER_MASK) << ((pin)*2))
+/**@}*/
+
+/* GPIO Output Mask Register Definitions. This register may be used to lock the output value of
+ * a pin as changes to masked pins will have no effect. */
+/** @defgroup gpio_outmask_regs GPIO OUTMASK register mapping.
```

Comment:
You don't need to close and open comments here.  Just put a second * at the start.  That way you actually get the description in the output.  Have you actually been running "make doc" and looking at how this all comes out?  (and checking the warnings in the logs?

Reply:
Okay...made changes in `ccs.h` and `gpio.h` for this. I like the continuous block better, so will do that in the future.